packages/data/src/hypoCompress.ts

```diff
@@ -0,0 +1,402 @@
+  const numberOfPlayers = data.players.length;
+  if (numberOfPlayers < 2 || numberOfPlayers > 6) {
+    return "";
+  }
+  out += `${data.players.length}`;
+
+  // deck
+  const [deck, okDeck] = compressDeck(data.deck);
+  if (!okDeck) {
+    return "";
+  }
+  out += `${deck},`;
+
+  // actions
+  const actions = compressActions(data.actions);
+  out += `${actions},`;
+
+  // variant id
+  const variant = variantsJSONArray.find(
+    (v) => v.name === data.options.variant,
+  );
+  if (variant === undefined) {
+    return "";
+  }
+  out += `${variant.id}`;
+
+  // add hyphens every 20 characters for URL posting (hyphens make the text wrap)
+  out = out.match(/.{1,20}/g)!.join("-");
+
+  return out;
+}
+
+/**
+ * Decompresses a string into a GameJSON object.
+ * Returns empty string if decompression fails.
+ *
+ * @param data The compressed string
+ */
+function gameJSONDecompress(data: string): string {
+  // remove all hyphens from URL
+  const normal = data.replace(/-/g, "");
+
+  // The compressed string is composed of 3 substrings, separated by comma.
+  const [deckString, actionsString, variantString] = [...normal.split(",", 3)];
+  const numPlayers = parseInt(deckString.charAt(0), 10);
+
+  const players = getPlayers(numPlayers);
+  if (players.length === 0) {
+    return "";
+  }
+
+  const [deck, okDeck] = decompressDeck(deckString.substring(1));
+  if (!okDeck) {
+    return "";
+  }
+
+  const actions = decompressActions(actionsString);
+
+  const variant = variantsJSONArray.find(
+    (v) => v.id.toString() === variantString,
+  )?.name;
+  if (variant === undefined) {
+    return "";
+  }
+
+  const original: GameJSON = {
+    players,
+    deck,
+    actions,
+    options: {
+      variant,
+    },
+    characters: [],
+    id: 0,
+    notes: [],
+    seed: "",
+  };
+
+  return JSON.stringify(original);
+}
+
+/**
+ * Compresses an array of DeckCard into a string.
+ * The first two characters of the string are numbers containing the min and max values of DeckCard.rank.
+ * The rest of the string contains series of strings one characters long, each representing a DeckCard.
+ *
+ * @param deck The array of DeckCard
+ */
+function compressDeck(deck: DeckCard[]): [string, boolean] {
+  const range = getRankMinMax(deck);
+  if (isMinMaxInvalid(range)) {
+    return ["", false];
+  }
+
+  let out = `${range.min}${range.max}`;
+  deck.forEach((card) => {
+    out += deckCardToString(card, range);
+  });
+
+  return [out, true];
+}
+
+/**
+ * Decompresses a string into an array of DeckCard.
+ * The first two characters of the string must be numbers containing the min and max values of DeckCard.rank.
+ * The rest of the string must be composed of series of strings one character long, each representing a DeckCard.
+ *
+ * @param src The compressed string
+ */
+function decompressDeck(src: string): [DeckCard[], boolean] {
+  const deck: DeckCard[] = [];
+  const range: MinMax = {
+    min: parseInt(src.charAt(0), 10),
```

Comment:
use parseintsafe instead of parseint, also check for isNaN, also use SLR

Reply:
Why use parseIntSafe since I'm only parsing a single character string?

Reply:
because parseint sucks and you should never use it

Reply:
marked as resolved but wasn't fixed